# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowcreator/nsIWindowCreator.idl
# Commit: fde2d124a1b1
# Full Hash: fde2d124a1b15877570390c7c69f5fcf6770640d
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-08 03:36:50
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 7.2: Create and use nsOpenWindowInfo types in nsWindowWatcher logic, r=kmag
#   
#   This patch builds on top of part 7.1 by creating this object within
#   nsWindowWatcher and ContentParent to carry the relevant information through
#   provider interfaces when creating new content windows. The nsOpenWindowInfo
# ==============================================================================

diff -r 7ccbbb127b7f -r fde2d124a1b1 toolkit/components/windowcreator/nsIWindowCreator.idl
--- a/toolkit/components/windowcreator/nsIWindowCreator.idl	Tue Apr 07 21:39:24 2020 +0000
+++ b/toolkit/components/windowcreator/nsIWindowCreator.idl	Tue Apr 07 21:39:32 2020 +0000
@@ -15,6 +15,7 @@
 
 #include "nsISupports.idl"
 
+interface nsIOpenWindowInfo;
 interface nsIRemoteTab;
 interface nsIURI;
 interface nsIWebBrowserChrome;
@@ -30,12 +31,8 @@
                     the parent, if any (and if the concept applies
                     to the underlying OS).
       @param chromeFlags Chrome features from nsIWebBrowserChrome
-      @param aOpeningTab The RemoteTab that is trying to open this new chrome
-                         window. Can be nullptr.
-      @param aOpener The window which is trying to open this new chrome window.
-                     Can be nullptr
-      @param aNextRemoteTabId The integer ID of the next remote tab actor to use.
-                              0 means there is no next remote tab ID to use.
+      @param aOpenWindowInfo Information used to open initial content in
+                             the new chrome window. Can be nullptr.
       @param cancel Return |true| to reject window creation. If true the
                     implementation has determined the window should not
                     be created at all. The caller should not default
@@ -44,9 +41,7 @@
   */
   nsIWebBrowserChrome createChromeWindow(in nsIWebBrowserChrome parent,
                                          in uint32_t chromeFlags,
-                                         in nsIRemoteTab aOpeningTab,
-                                         in mozIDOMWindowProxy aOpener,
-                                         in unsigned long long aNextRemoteTabId,
+                                         in nsIOpenWindowInfo aOpenWindowInfo,
                                          out boolean cancel);
 };
 